### PR TITLE
minor refactoring before changes

### PR DESCRIPTION
**Pull request by ryanfkeepers**
renaming structs so that they don't follow the interface naming conventions, updating and expanding the test suite setup for api.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3562

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by meain** [comment]
This could probably use a rebase before review as there are quite a few changes from previous PRs.

**Comment by ryanfkeepers** [comment]
> This could probably use a rebase before review as there are quite a few changes from previous PRs.

Yep, planning on it.  CI flakes have made it troublesome to get the prior PR through.  I'll probably do a rebase sweep from here to the tail once that gets merged.

File: src/pkg/services/m365/api/drive.go [hunk 1]
```diff
@@ -305,3 +306,13 @@ func (c Drives) DeleteItemPermission(
...
 }
+
+// DriveItemCollisionKeyy constructs a key from the item name.
+// collision keys are used to identify duplicate item conflicts for handling advanced restoration config.
+func DriveItemCollisionKey(item models.DriveItemable) string {
+	if item == nil {
+		return ""
+	}
+
+	return ptr.Val(item.GetName())
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/drive.go:317]: is the cache going to be per drive or per folder? If it's per drive the collision key will need to include the full (display name) path of the item

**Review by ashmrtn — APPROVED** [review]

File: src/pkg/services/m365/api/drive.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/drive.go:317]: It will be per folder.

**Pull request merged**